accepted/future-releases/primary-constructors/feature-specification.md

```diff
@@ -840,15 +863,16 @@ positional or named parameter remains optional; if it has a default value
   unchanged from _L_ to _L2_ *(this is a plain, non-declaring parameter)*.
 - Otherwise, a formal parameter (named or positional) of the form `var T p`
   or `final T p` where `T` is a type and `p` is an identifier is replaced
-  in _L2_ by `this.p`, along with its default value, if any. Next, a
-  semantic instance variable declaration corresponding to the syntax `T p;`
-  or `final T p;` is added to _D2_. It includes the modifier `final` if the
-  parameter in _L_ has the modifier `final` and _D_ is not an `extension
-  type` decaration; if _D_ is an `extension type` declaration then the name
-  of `p` specifies the name of the representation variable. In all cases, if
-  `p` has the modifier `covariant` then this modifier is removed from the
-  parameter in _L2_, and it is added to the instance variable declaration
-  named `p`.
+  in _L2_ by `this.p`, along with its default value, if any. If the
+  parameter has the modifier `var` and _D_ is an extension type declaration
+  then a compile-time error occurs. Otherwise, a semantic instance variable
+  declaration corresponding to the syntax `T p;` or `final T p;` is added
+  to _D2_. It includes the modifier `final` if the parameter in _L_ has the
+  modifier `final` and _D_ is not an `extension type` decaration; if _D_ is
```

Comment:
Move the "and _D_ is not an `extension type declaration` to the prior sentence. The entire variable declaration is not introduced if it's an extension type declaration, not just the `final`.